cmd/preview/cmdpreview.go

```diff
@@ -117,18 +121,14 @@ func (r *PreviewRunner) RunE(cmd *cobra.Command, args []string) error {
 			ForceConflicts:  false,
 			FieldManager:    common.DefaultFieldManager,
 		}
-		ch = r.Applier.Run(ctx, objs, apply.Options{
+		ch = r.Applier.Run(ctx, inventory.WrapInventoryObj(inv), objs, apply.Options{
```

Comment:
I don't believe this will work. There are two versions of `WrapInventoryObj`; one for ConfigMap and one for `ResourceGroup`. We will need to set this `InventoryFactoryFunc` based on the inventory object that is detected, so we can correctly handle both `ConfigMap` and `ResourceGroup` inventory objects while we're migrating. This is currently being worked on.

Reply:
Thanks for pointing this out. I updated it to use the function from the `provider.InventoryClient`. In addition to the `InventoryFactoryFunc`, also added two other functions to change `InventoryInfo` to `Unstructured` and vice versa.